tests/RobotFramework/tests/tedge_write.robot

```diff
@@ -33,53 +67,93 @@ Preserves permissions if a destination exists
+    Execute Command as user    tedge
+    ...    umask ${umask} && echo abc | sudo tedge-write ${dest_filename}    strip=True
+    
+    Should Have File Ownership    ${dest_filename}    root:root
 
+    There should be no leftover temporary files    ${dest_filename}
 
 Changes permissions if a destination doesn't exist
     [Arguments]          ${umask}
 
-    ${dest_filename}=    Execute Command    mktemp --dry-run    strip=True
-    Execute Command      umask ${umask} && echo abc | sudo tedge-write ${dest_filename} --user tedge --group tedge --mode 700
+    ${dest_filename}=    Create a temporary file    dry_run=${True}
+    Execute Command as user    tedge
+    ...    umask ${umask} && echo abc | sudo tedge-write ${dest_filename} --user tedge --group tedge --mode 700
+    
+    Should Have File Ownership    ${dest_filename}    tedge:tedge
+    Should Have File Mode         ${dest_filename}    700
 
-    Execute Command      ls ${dest_filename}
-    ${user_owner}=       Execute Command    stat -c '%U' ${dest_filename}    strip=True
-    Should Be Equal      ${user_owner}    tedge
-    ${group_owner}=      Execute Command    stat -c '%G' ${dest_filename}    strip=True
-    Should Be Equal      ${group_owner}    tedge
-    ${mode}=      Execute Command    stat -c '%a' ${dest_filename}    strip=True
-    Should Be Equal      ${mode}    700
+    There should be no leftover temporary files    ${dest_filename}
 
 Preserves permissions if a destination exists
     [Arguments]          ${umask}
 
-    ${dest_filename}=    Execute Command    mktemp    strip=True
-    Execute Command      chown tedge:tedge ${dest_filename}
-    Execute Command      chmod 666 ${dest_filename}
+    ${dest_file}=    Create a temporary file
+    Execute Command      chown tedge:tedge ${dest_file}
+    Execute Command      chmod 666 ${dest_file}
+
+    Execute Command as user    tedge
+    ...    umask ${umask} && echo abc | sudo tedge-write ${dest_file} --user root --group root --mode 700
+    
+    Should Have File Ownership    ${dest_file}    tedge:tedge
+    Should Have File Mode         ${dest_file}    666
 
-    Execute Command      umask ${umask} && echo abc | sudo tedge-write ${dest_filename} --user root --group root --mode 700
-    Execute Command      ls ${dest_filename}
-    ${user_owner}=       Execute Command    stat -c '%U' ${dest_filename}    strip=True
-    Should Be Equal      ${user_owner}    tedge
-    ${group_owner}=      Execute Command    stat -c '%G' ${dest_filename}    strip=True
-    Should Be Equal      ${group_owner}    tedge
-    ${mode}=      Execute Command    stat -c '%a' ${dest_filename}    strip=True
-    Should Be Equal      ${mode}    666
+    There should be no leftover temporary files    ${dest_file}
 
 Custom Setup
     Setup    skip_bootstrap=True
     Execute Command    ./bootstrap.sh --no-bootstrap --no-connect
-    Execute Command    sudo --user\=tedge bash
+
+    Create a regular user
+
+Create a temporary file
```

Comment:
The Create a temporary file keyword should include cleanup logic for removing temporary directories and files created during the tests to avoid polluting the test environment, especially if the tests fail

Reply:
fixed in 0dff38f0